safe-global/safe-wallet-web
# fix: `useAsync` dependency arrays

## What it solves

Resolves #2061

## How this PR fixes it

This adds the exhaustive dependency array linter to our `useAsync` hook. All relative findings have been adjusted accordingly.

Dependency arrays that needed changing were found in:

1. `DecodedTx`
2. `useRedefine`
3. `useGasPrice`
4. `useIsValidExecution`
5. `useWalletCanRelay`

## How to test it

According to the above list:

1. Observe the transaction info displaying in the sign/execute form (above the balance change).
2. Redefine works.
3. Gas estimation works in Safe creation.
4. Invalid executions are warned about in the execution form ("This transaction will most likely fail...").
5. It is _not_ possible to relay 6 times in an hour.

## Screenshots

## Checklist
